core/src/style/textStyleBuilder.cpp

```diff
@@ -322,9 +501,7 @@ TextStyle::Parameters TextStyleBuilder::applyRule(const DrawRule& _rule,
             p.labelOptions.priority = (float)priority;
         }
         _rule.get(StyleParamKey::text_collide, p.labelOptions.collide);
-        if (!_rule.get(StyleParamKey::text_interactive, p.interactive)) {
-            _rule.get(StyleParamKey::interactive, p.interactive);
-        }
+        _rule.get(StyleParamKey::text_interactive, p.interactive);
```

Comment:
`interactive` parameter is inherited from the parent `point` style, if `text` style itself is not having one defined.